rust/src/builder/mod.rs

```diff
@@ -37,6 +37,8 @@ enum BuilderError {
     Required(String),
     #[error("Failed to find valid credential.")]
     MissingCredential,
+    #[error("Failed to decode sas key. {0}")]
```

Comment:
Perhaps it's worth adding more detail to the error message:

```suggestion
    #[error("Failed to decode SAS key: {0}\nSAS keys must be percent-encoded. They come encoded in the Azure portal and Azure Storage Explorer.")]
```